app/models/mentee_application_state.rb

```diff
@@ -32,4 +32,12 @@ class MenteeApplicationState < ApplicationRecord
     accepted: 5,
     rejected: 6
   }
+
+  class << self
+    # state machine for mentee application
+    def next(status:)
+      return if %i[accepted rejected].include?(status)
+      statuses.keys[statuses.keys.index(status.to_s) + 1]
+    end
+  end
 end
```

Comment:
Curious to know why you went for this type of definition over just creating a class method?

Reply:
@toyhammered I was thinking we could add the ability to toggle the  buttons text to reflect the next stage. So it would say something like this: "Promote to 'Third Stage'" or if application is at final stage, "Accept Application" this could. If we agree on that feature, I think the Class method would be needed to be able to call it from the view component

Reply:
wouldn't the view component already be using the `user_mentee_state` record??

Reply:
Also both methods behind the scenes should be using common code anyways so we can offer both flavors if needed.